packages/wallet/src/test-utils/launchNode.ts

```diff
@@ -0,0 +1,123 @@
+import { BaseAssetId } from '@fuel-ts/address/configs';
+import { Provider } from '@fuel-ts/providers';
+import { spawn } from 'child_process';
+import fs from 'fs/promises';
+
+import type { WalletUnlocked } from '../wallets';
+
+import { defaultChainConfig } from './defaultChainConfig';
+import { generateTestWallet } from './generateTestWallet';
+
+const defaultFuelCoreArgs = ['--vm-backtrace', '--utxo-validation', '--manual_blocks_enabled'];
+
+type LaunchNodeOptions = {
+  chainConfigPath: string;
+  consensusKey: string;
+  dbPath: string;
+  args?: string[];
+};
+
+/**
+ * Launches a fuel-core node.
+ * @param chainConfigPath - path to the chain configuration file.
+ * @param consensusKey - the consensus key to use.
+ * @param dbPath - the path to the folder to use for the fuel-core db.
+ * @param args - additional arguments to pass to fuel-core
+ * */
+export const launchNode = async ({
+  chainConfigPath,
+  consensusKey,
+  dbPath,
+  args = defaultFuelCoreArgs,
+}: LaunchNodeOptions): Promise<() => void> =>
+  new Promise((resolve) => {
+    // This string is logged by the client when the node has successfully started. We use it to know when to resolve.
+    const graphQLStartSubstring = 'Binding GraphQL provider to';
+
+    const child = spawn('fuel-core', [
+      'run',
+      '--db-path',
+      dbPath, // hardcoded
+      '--consensus-key',
+      consensusKey,
+      '--chain',
+      chainConfigPath,
+      ...args,
+    ]);
+
+    // Cleanup function where fuel-core is stopped.
+    const cleanup = () => {
+      child.kill('SIGINT');
+
+      // Remove all the listeners we've added.
+      child.stdout.removeAllListeners();
+      child.stderr.removeAllListeners();
+    };
+
+    child.stderr.setEncoding('utf8');
+
+    // Look for a specific graphql start point in the output.
+    child.stderr.on('data', (chunk: string) => {
+      // Look for the graphql service start.
+      if (chunk.indexOf(graphQLStartSubstring) !== -1) {
+        // Resolve with the cleanup method.
+        resolve(cleanup);
+      }
+    });
+
+    // Process exit.
+    process.on('exit', cleanup);
+
+    // Catches ctrl+c event.
+    process.on('SIGINT', cleanup);
+
+    // Catches "kill pid" (for example: nodemon restart).
+    process.on('SIGUSR1', cleanup);
+    process.on('SIGUSR2', cleanup);
+
+    // Catches uncaught exceptions.
+    process.on('uncaughtException', cleanup);
+  });
+
+const generateWallets = async (count: number, provider: Provider) => {
+  const wallets: WalletUnlocked[] = [];
+  for (let i = 0; i < count; i += 1) {
+    const wallet = await generateTestWallet(provider, [[1_000, BaseAssetId]]);
+    wallets.push(wallet);
+  }
+  return wallets;
+};
+
```

Comment:
```suggestion
const generateWallets = async (count: number, provider: Provider) => {
  const calls: Promise<WalletUnlocked>[] = [];
  for (let i = 0; i < count; i += 1) {
    calls.push(generateTestWallet(provider, [[1_000, BaseAssetId]]));
  }
  return Promise.all(calls);
};
```

This parallellizes wallet generation so that there is no need to wait for one wallet to be generated to start with the generation of the next one.

Reply:
@nedsalk I just tried this out and the client is reporting some UTXO-related error: `Transaction is not inserted. UTXO is not existing: 0x000000000000000000000000000000000000000000000000000000000000000002: {"response":{"data":null,"errors":[{"message":"Transaction is not inserted. UTXO is not existing:`

Looks like it's not a good idea to try and generate wallets in parallel 🤔

Reply:
Hmm, in theory it should work and it would be handy to have it, especially during testing if we want to run one node and have each test use an unique wallet to avoid coupling tests, in which case we'd need _a lot_ of wallets. It also seems like an interesting problem to investigate. Could you please just create an issue after you merge this PR?

Reply:
@nedsalk sure, I'll file a follow-up

Reply:
@Dhaiwat10 @nedsalk The cause of this is the flow process used to generate a new test wallet. Basically, the steps are generating a new wallet and using UTXOs from the Genesis wallet to transfer funds to it. 

The process to fund this new wallet is basically fetching UTXOs (Unspent Tokens) from the Genesis wallet and then creating a transfer transaction to this new wallet. 

The change from this transaction (i.e., the new UTXOs) becomes available to the Genesis wallet only after the transaction has been fully processed, and the used UTXOs on the transaction, are now spent.

So when you try to execute all loops concurrently you are creating a race condition. Many of these processes are going to use the same UTXOs to create the transfer transaction, and the transaction that is processed first will succeed but all other ones are going to fail since they are now using a token that was spent. 

This explains the error message: `UTXO is not existing.`